File: platform/frontend/src/api/hybridvalidation.ts
import { authFetch } from './auth'

const API = import.meta.env.VITE_API_URL || ''

export interface HybridValidation {
  id: number
  projectName: string
  operationType: string
  aiOutput: string
  validationResult: string
  retryCount: number
  maxRetries: number
  usedFallback: boolean
  fallbackAction: string
  failureReason: string
  rulesApplied: string
  rulesPassedCount: number
  rulesFailedCount: number
  confidenceScore: number
  validationTimeMs: number
  status: string
  createdAt: string
}

export interface ValidateResponse {
  id: number
  projectName: string
  operationType: string
  validationResult: string
  retryCount: number
  usedFallback: boolean
  fallbackAction: string
  failureReason: string
  rulesPassedCount: number
  rulesFailedCount: number
  confidenceScore: number
  validationTimeMs: number
  rulesApplied: string[]
  passedRules: string[]
  failedRules: string[]
  recommendation: string
}

export interface OperationType {
  id: string
  name: string
  description: string
  severity: string
  rules: string[]
}

export interface ValidationStats {
  total: number
  byType: { type: string; count: number; avgConfidence: number }[]
}

export async function listValidations(): Promise<HybridValidation[]> {
  const res = await authFetch(`${API}/api/hybrid-validation`)
  if (!res.ok) return []
  return res.json()
}

export async function validateOperation(projectName: string, operationType: string, aiOutput: string): Promise<ValidateResponse> {
  const res = await authFetch(`${API}/api/hybrid-validation/validate`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ projectName, operationType, aiOutput })
  })
  return res.json()
}

export async function deleteValidation(id: number): Promise<void> {
  await authFetch(`${API}/api/hybrid-validation/${id}`, { method: 'DELETE' })
}

export async function getValidationStats(): Promise<ValidationStats> {
  const res = await authFetch(`${API}/api/hybrid-validation/stats`)
  if (!res.ok) return { total: 0, byType: [] }
  return res.json()
}

export async function getOperationTypes(): Promise<OperationType[]> {
  const res = await fetch(`${API}/api/hybrid-validation/operations`)
  if (!res.ok) return []
  return res.json()
}

File: platform/backend/AiDevRequest.API/Entities/HybridValidation.cs
namespace AiDevRequest.API.Entities;

public class HybridValidation
{
    public int Id { get; set; }
    public string UserId { get; set; } = string.Empty;
    public string ProjectName { get; set; } = string.Empty;
    public string OperationType { get; set; } = string.Empty; // db-migration, git-operation, file-operation, api-validation, security-check
    public string AiOutput { get; set; } = string.Empty;
    public string ValidationResult { get; set; } = string.Empty; // passed, failed, retried
    public int RetryCount { get; set; }
    public int MaxRetries { get; set; } = 3;
    public bool UsedFallback { get; set; }
    public string FallbackAction { get; set; } = string.Empty;
    public string FailureReason { get; set; } = string.Empty;
    public string RulesApplied { get; set; } = string.Empty; // JSON array of rule names
    public int RulesPassedCount { get; set; }
    public int RulesFailedCount { get; set; }
    public double ConfidenceScore { get; set; }
    public double ValidationTimeMs { get; set; }
    public string Status { get; set; } = "active";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
